### PR TITLE
Allow hyphens in locale name

### LINKED ISSUE #9446: Localization fails if language does not match locale
In site configuration, if you do this:

```
[languages.es-ES]
```

then dates, currencies, numbers, and percentages will not be localized because `es-ES` does not match any of the locales in [gohugoio/localescompressed](https://github.com/gohugoio/localescompressed). The locales use underscores instead of hyphens.

See <https://github.com/gohugoio/localescompressed/pull/1>

Somewhat related to <https://github.com/gohugoio/hugo/issues/9109>.

### PR DESCRIPTION
**Pull request by jmooring**
Fixes gohugoio/hugo#9446

### CODE REVIEW

**Comment by bep** [comment]
OK, let's start over, I was a little fast earlier. Why does this change require change in 4 files? I suspect some of them are formatting changes, but with thousands of lines changed, it's not possible for me to decide.

File: locales_test.go [hunk 1]
```diff
@@ -52,6 +52,12 @@ func TestGetTranslator(t *testing.T) {
 		c.Assert(tnn.MonthWide(d.Month()), qt.Equals, "januar")
 	})
 
+	c.Run("Basic", func(c *qt.C) {
```
Comment (bep) [inline locales_test.go]: Make the test name distinct.

**Comment by jmooring** [comment]
> Why does this change require change in 4 files?

```text
currencies.autogen.go  --> auto-generated by gen/main.go
locales.autogen.go     --> auto-generated by gen/main.go
locales.go
locales_test.go
```

File: locales_test.go (hunk 1 above)
Comment (jmooring) [reply locales_test.go]: Done

**Comment by bep** [comment]
If you can narrow down this change set to the 1 file (+ test) that is touched by this change, I will merge it.

And yes, I understand your point, but dont't have the time to figure out why your go generate generates different from mine, and I certainly cannot spend time on the 10K line diff.

**Comment by bep** [comment]
OK, I guess the go generate script does not apply `gofmt`, which would explain the diff, but I still don't see why you would need to regenerate these files for this particular patch.

**Comment by jmooring** [comment]
> I still don't see why you would need to regenerate these files

I apologize. I didn't see any harm in doing so. I have reverted those changes.

**Comment by bep** [comment]
>I apologize. I didn't see any harm in doing so. I have reverted those changes.

The bigger the diff, the harder is to merge.

**Pull request merged**